### PR TITLE
Model

### PR DESCRIPTION
**Pull request by annafrei924**

### CODE REVIEW

**Comment by annafrei924** [comment]
Model with elements of controller also completed. Not sure why it's not passing checkstyles. I fixed the comments and changed it but it's showing failures on old code.

File: src/main/java/bfk/brickbreaker/Ball.java [hunk 1]
```diff
@@ -3,28 +3,78 @@
...
+    //checks if ball collides with a wall
 
+    public boolean collides() {
+        if (x + radius >= width || x - radius <= 0
+                ||  y - radius <= 0 || y + radius >= height) {
+            return true;
+        }
+        else {
+            return false;
+        }
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/Ball.java]: This can be simplified

File: src/main/java/bfk/brickbreaker/BBController.java [hunk 2]
```diff
@@ -1,4 +1,88 @@
...
+                    direction = LEFT;
+                } else if (newX + radius >= ball.getWidth()) {
+                    direction = RIGHT;
+                } else if (newY - radius <= 0) {
+                    direction = TOP;
+                } else if (newY + radius >= ball.getHeight()) {
+                    timer.stop();
+                }
+
+                if (direction != NONE) {
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/BBController.java]: The if statement here is unnecessary

File: src/main/java/bfk/brickbreaker/Brick.java [hunk 3]
```diff
@@ -6,27 +6,67 @@
...
+        return rows;
     }
 
     private void populateBricks() {
-        Random rand = new Random();
+        //randomly populate top row with new bricks
         for (int col = 0; col < getWidth(); col++) {
-            if (rand.nextDouble() < 1.0 / 3.0) {
-                field[0][col] = 1;
+            if (rand.nextDouble() < 1.0 / 2.0) {
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/Brick.java]: nextBoolean()

File: src/main/java/bfk/brickbreaker/Direction.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package bfk.brickbreaker;
+
+public enum Direction {
+    RIGHT,
+    LEFT,
+    TOP,
+    PADDLE,
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/Direction.java]: Paddle is a direction?

File: src/main/java/bfk/brickbreaker/Paddle.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    public Paddle(int width, int height) {
+        this.width = width;
+        this.height = height;
+        this.x = x;
+        this.y = y;
+    }
+
+
+    public void movePanel(MouseEvent e, Boolean isMoving, int changeX, int changeY) {
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/Paddle.java]: Why is this called movePanel?

File: src/main/java/bfk/brickbreaker/Paddle.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        if (isMoving) {
+            int currX = e.getXOnScreen();
+            int currY = e.getYOnScreen();
+
+            setLocation(currX - changeX, currY - changeY);
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/Paddle.java]: The y value of the paddle doesn't change

File: src/main/java/bfk/brickbreaker/Direction.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package bfk.brickbreaker;
+
+public enum Direction {
```
Comment (andrewoid) [inline src/main/java/bfk/brickbreaker/Direction.java:3]: These are collision directions?

Not sure what `BOTTOMPADDLE` means

Comment (annafrei924) [reply src/main/java/bfk/brickbreaker/Direction.java:3]: It hits the paddle on the bottom. Trying to differentiate between that and when it hits the actual bottom (falls off the screen & the user 'loses' or moves on to the next round)

**Pull request closed without merging**